operator/internal/manifests/build.go

```diff
@@ -61,12 +61,16 @@ func BuildAll(opts Options) ([]client.Object, error) {
 	res = append(res, BuildLokiGossipRingService(opts.Name))
 
 	if opts.Stack.Rules != nil && opts.Stack.Rules.Enabled {
-		rulesCm, err := RulesConfigMap(&opts)
+
+		rulesCMShards, err := RulesConfigMapShards(&opts)
 		if err != nil {
 			return nil, err
 		}
 
-		res = append(res, rulesCm)
+		for _, shard := range rulesCMShards {
+			opts.RulesConfigMapNames = append(opts.RulesConfigMapNames, shard.Name)
+			res = append(res, shard)
+		}
```

Comment:
```suggestion
               var configMapNames []string
		for _, shard := range rulesCMShards {
			configMapNames = append(configMapNames, shard.Name)
			res = append(res, shard)
		}
		opts.RulesConfigMapNames  = configMapNames
```

Reply:
@btaani I think I was forgetting if `opts` is always new each time it got into `Build` method or if it was possibly to continuously add the same names between calls. So in light of that, opted to for the scenario where it's just always assigns a fresh list `opts.RulesConfigMapNames`. 

However, I think the existing method works as well, so what I suggested might just be overkill/overcautious. Checking the code, `opts` is new every time so the worry I had is moot.